roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -44,26 +44,49 @@ class RooMinimizer : public TObject {
 public:
    enum class FcnMode { classic, gradient, generic_wrapper };
 
+   /// Config argument to RooMinimizer ctor
+   struct Config {
+      double recoverFromNaN = 10.; // RooAbsMinimizerFcn config
+      int optConst = 0;            // RooAbsMinimizerFcn config
+      int printEvalErrors = 10;    // RooAbsMinimizerFcn config
+      int doEEWall = 1;            // RooAbsMinimizerFcn config
+      int offsetting = -1;         // RooAbsMinimizerFcn config
+      const char *logf = nullptr;  // RooAbsMinimizerFcn config
+      int nWorkers = getDefaultWorkers(); // RooAbsMinimizerFcn config that can only be set in ctor
+      int parallelGradient = 0;          // RooAbsMinimizerFcn config that can only be set in ctor
+      int parallelLikelihood = 0;        // RooAbsMinimizerFcn config that can only be set in ctor
+      int verbose = 0;                    // local config
+      bool profile;                       // local config
+      std::string minimizerType = "";     // local config
+      FcnMode fcnMode = FcnMode::classic; // local config
+   private:
+      int getDefaultWorkers();
+   };
+
    explicit RooMinimizer(RooAbsReal &function, FcnMode fcnMode = FcnMode::classic);
-   explicit RooMinimizer(
-      std::shared_ptr<RooFit::TestStatistics::RooAbsL> likelihood,
-      RooFit::TestStatistics::LikelihoodMode likelihoodMode = RooFit::TestStatistics::LikelihoodMode::serial,
-      RooFit::TestStatistics::LikelihoodGradientMode likelihoodGradientMode =
-         RooFit::TestStatistics::LikelihoodGradientMode::multiprocess);
+   explicit RooMinimizer(RooAbsReal &function, Config &cfg);
 
    ~RooMinimizer() override;
```

Comment:
Thanks for deleting the definition of the custom `~RooMinimizer` destructor!

You should also delete de declaration in the header file though :)